### PR TITLE
[embind] Change GenericEnumValue's type to int

### PR DESCRIPTION
**Pull request by aheejin**
`long` is interpreted as a pointer and causes negative `enum class` values to be casted to unsigned and become large overflown values when `CAN_ADDRESS_2GB` is true:
https://github.com/emscripten-core/emscripten/blob/0a3b7539856df97370c2e42e174850883c735c9a/src/jsifier.js#L153

### CODE REVIEW

File: test/test_other.py [hunk 1]
```diff
@@ -14298,3 +14298,38 @@ def test_errar(self):
   def test_no_input_files(self):
     err = self.expect_fail([EMCC, '-c'])
     self.assertContained('clang: error: no input files', err)
+
+  def test_negative_enum_values(self):
```
Comment (sbc100) [inline test/test_other.py:14302]: test_embind_negative_enum_values?

File: test/test_other.py [hunk 2]
```diff
@@ -14298,3 +14298,38 @@ def test_errar(self):
...
+
+      EMSCRIPTEN_BINDINGS(utility) {
+          enum_<value>("value")
+              .value("neg", value::neg)
+              .value("zero", value::zero)
+              .value("pos", value::pos);
+      }
+    '''
+    expected = '-1\n0\n1\n'
+    self.do_run(src, assert_identical=True, expected_output=expected,
```
Comment (sbc100) [inline test/test_other.py:14334]: I think you can drop the `assert_identical` here.. if there happens for be some output before or after we don't particularly care in this case I think?

**Review by sbc100 — APPROVED** [review]

File: test/test_other.py (hunk 1 above)
Comment (aheejin) [reply test/test_other.py]: Done

File: test/test_other.py (hunk 2 above)
Comment (aheejin) [reply test/test_other.py]: Done

**Pull request merged**